libs/gl-client/src/pairing/new_device.rs

```diff
@@ -0,0 +1,144 @@
+use super::PairingSessionData;
+use crate::{
+    credentials::{Device, TlsConfigProvider},
+    pb::scheduler::{pairing_client::PairingClient, PairDeviceRequest, PairingQr},
+    tls::{self, TlsConfig},
+};
+use log::debug;
+use tokio::sync::mpsc;
+use tonic::transport::Channel;
+
+type Result<T, E = super::Error> = core::result::Result<T, E>;
+
+pub struct Unconnected();
+pub struct Connected(PairingClient<Channel>);
+
+pub struct Client<T> {
+    inner: T,
+    uri: String,
+    tls: TlsConfig,
+}
+
+impl Client<Unconnected> {
+    pub fn new<T>(creds: T) -> Client<Unconnected>
+    where
+        T: TlsConfigProvider,
```

Comment:
Should we return an error if a `device_cert` is passed in?

Reply:
Good question, what would you prefer?